runtime/src/mm/cpp/ThreadState.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include "ThreadData.hpp"
+#include "ThreadState.hpp"
+
+using namespace kotlin;
+
+namespace {
+
+// TODO: Can we make it constexpr? Is it worthwhile at all?
+bool isStateSwitchAllowed(mm::ThreadState oldState, mm::ThreadState newState) {
+    switch (oldState) {
```

Comment:
Seems awkward. What's wrong with `return oldState != newState`?

Reply:
I suppose that we will have more than 2 states and this is just a template for a more complex check. But currently it's indeed just `oldState != newState`. Fixed.